contracts/StructHash.cairo

```diff
@@ -0,0 +1,114 @@
+%lang starknet
+%builtins pedersen range_check ecdsa bitwise
+
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.cairo.common.hash_state import (
+    HashState, hash_finalize, hash_init, hash_update, hash_update_single)
+from starkware.cairo.common.registers import get_fp_and_pc
+
+struct StarkNet_Domain:
+    member name : felt
+    member version : felt
+    member chain_id : felt
+end
+
+struct Person:
+    member name : felt
+    member wallet : felt
+end
+
+struct Mail:
+    member from2 : Person
+    member to : Person
+    member contents : felt
+end
+
+# H('StarkNetDomain(felt name,felt version,felt chainId)')
```

Comment:
Given function declaration in Cairo we should do ` H('StarkNetDomain(name:felt,version:felt,chainId:felt)')` to be consistent. That's also how it is defined in https://github.com/argentlabs/argent-x/discussions/14.

And the same for the other struct definitions (Mail and Person).